src/server/job_func.c

```diff
@@ -516,6 +516,8 @@ job_free(job *pj)
 			if (pwt->wt_type == WORK_Deferred_Reply) {
 				tbr = (struct batch_request *)pwt->wt_parm1;
 				if (tbr != NULL) {
+					/* wt_event2 either has additional data (like msgid) or NULL */
+					free(pwt->wt_event2);
```

Comment:
@subhasisb in case of RPP mode wt_aux2 is set instead of wt_event2. This may not be the usual scenario but it can happen.